### PR TITLE
Split the project parsing and the project into separate classes

### PR DESCRIPTION
**Pull request by davidfowl**
- Added ProjectReader

/cc @anurse conflicts a' coming

### CODE REVIEW

**Comment by rynowak** [comment]
It would be really handy if this was available for `TestHost` to use

**Comment by anurse** [comment]
Travis? TRAVIS?! TRAVIS?!!

**Comment by anurse** [comment]
Someone told me not to review code unless it passes Travis sooooooo,... :trollface:

**Comment by davidfowl** [comment]
I agree with that actually the red is bugging me so much :cry: . But mac osx is failing because of relative path and tmp crap. I'll fix that as soon as I get this in so we can be green again

**Comment by anurse** [comment]
Suuuuuure ;)

**Comment by davidfowl** [comment]
@anurse promise :smile:

File: src/Microsoft.Dnx.Runtime/Project.cs [hunk 1]
```diff
@@ -23,19 +19,19 @@ public class Project
 
         public static TypeInformation DefaultCompiler = DefaultRuntimeCompiler;
 
-        private readonly Dictionary<FrameworkName, TargetFrameworkInformation> _targetFrameworks = new Dictionary<FrameworkName, TargetFrameworkInformation>();
-        private readonly Dictionary<FrameworkName, CompilerOptions> _compilationOptions = new Dictionary<FrameworkName, CompilerOptions>();
-        private readonly Dictionary<string, CompilerOptions> _configurations = new Dictionary<string, CompilerOptions>(StringComparer.OrdinalIgnoreCase);
+        // REVIEW: It's kinda hacky making these internal but the reader needs to set them
```
Comment (anurse) [inline src/Microsoft.Dnx.Runtime/Project.cs]: Could you put them in a constructor? If you did that I think it'd be OK for the constructor to be public and the grossness would go away.

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/Project.cs]: The dictionaries yes but not the default compiler or target framework.

Comment (anurse) [inline src/Microsoft.Dnx.Runtime/Project.cs]: Why not? If they come from the JSON then they should be fine there. If they don't come from the JSON they don't need to be in ProjectReader :)

Comment (davidfowl) [inline src/Microsoft.Dnx.Runtime/Project.cs]: I'd have to create the project after parsing everything. Which might be possible but I'd have to look.

Comment (anurse) [inline src/Microsoft.Dnx.Runtime/Project.cs]: Fair enough for now.

**Pull request merged**